### PR TITLE
feat(testing): #1608 added validation for all project fields

### LINKED ISSUE #1608: Automated test part 4 - public project, test other columns/fields
Currently the project title is tested, we should test the other columns displayed too

- type
- description
- url
- repository
- monitors
- pings
- user
- last updated

### PR DESCRIPTION
**Pull request by webkhushboo**
closes #1608

### Notes

A summary of what was achieved in this PR

- [ ] Added validation for all public project fields 
- description
- url
- repository
- monitors
- pings
- user
- last updated

### CODE REVIEW

File: web/cypress.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "baseUrl": "http://localhost:4200",
+  "baseUrl": "https://pipelinedashboard-khush.firebaseapp.com",
```
Comment (eddiejaoude) [inline web/cypress.json]: this should be localhost, I will update

File: web/test/data/private-project.json [hunk 2]
```diff
@@ -1,6 +1,23 @@
 {
+  "uid": "test-private-project",
   "type": "private",
   "title": "Demo private project",
   "description": "Demo private project description added",
-  "url": "https://www.pipelinedashboard.io"
+  "_url": "https://www.pipelinedashboard.io",
```
Comment (eddiejaoude) [inline web/test/data/private-project.json]: why the underscore for  `url`?

File: web/test/data/private-project.json [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
+  "logoUrl": "",
+  "repositories": ["19deb070-1603-4fea-a774-6da2f93253ce", "97b2389a-4163-48b3-9fa8-ad8eb7937647"],
```
Comment (eddiejaoude) [inline web/test/data/private-project.json:8]: These `uid`s will specific to an environment. I think we have a new PR to improve these, so probably ok for now

File: web/test/integration/project.feature [hunk 4]
```diff
@@ -4,6 +4,13 @@ Scenario: List public project on the homepage
     Given there is a document "test-public-project" with the json "test/data/public-project.json" in collection "projects"
     And the "/" page is open
     And the text "Demo public project" is in the element ".project__list__title"
+    And the text "Demo public project description added" is in the element ".mat-column-description"
+    And the text "https://www.pipelinedashboard.io" is in the element ".project__list__url__content"
+    And the text "2" is in the element ".mat-column-repository"
+    And the text "1" is in the element ".mat-column-monitors"
+    And the text "0" is in the element ".mat-column-pings"
+    And the text "webkhushboo" is in the element ".mat-column-user"
+    And the text "1 second ago" is in the element ".mat-column-lastDate"
```
Comment (eddiejaoude) [inline web/test/integration/project.feature:14]: Need to use `Given` `Then` `When` with `And` to extend any step

File: web/test/integration/project.feature [hunk 5]
```diff
@@ -4,6 +4,13 @@ Scenario: List public project on the homepage
     Given there is a document "test-public-project" with the json "test/data/public-project.json" in collection "projects"
     And the "/" page is open
     And the text "Demo public project" is in the element ".project__list__title"
+    And the text "Demo public project description added" is in the element ".mat-column-description"
+    And the text "https://www.pipelinedashboard.io" is in the element ".project__list__url__content"
+    And the text "2" is in the element ".mat-column-repository"
+    And the text "1" is in the element ".mat-column-monitors"
+    And the text "0" is in the element ".mat-column-pings"
+    And the text "webkhushboo" is in the element ".mat-column-user"
```
Comment (eddiejaoude) [inline web/test/integration/project.feature]: I think we should use a test/fake user here

File: web/test/data/public-project.json [hunk 6]
```diff
@@ -1,6 +1,23 @@
...
+    "path": "/home",
+    "expectedCode": 200,
+    "expectedText": "",
+    "successfulPings": 0,
+    "unsuccessfulPings": 0
+  }],
+  "pings": [],
+  "tokens": [],
+  "views": 1,
+  "followers": 0
```
Comment (eddiejaoude) [inline web/test/data/public-project.json]: This document is missing a user?

File: web/test/data/private-project.json (hunk 2 above)
Comment (eddiejaoude) [reply web/test/data/private-project.json]: I will update

File: web/test/data/public-project.json (hunk 6 above)
Comment (eddiejaoude) [reply web/test/data/public-project.json]: I will update

File: web/test/integration/project.feature (hunk 5 above)
Comment (eddiejaoude) [reply web/test/integration/project.feature]: I will update

File: web/test/integration/project.feature (hunk 4 above)
Comment (eddiejaoude) [reply web/test/integration/project.feature:14]: I will update

**Review by eddiejaoude — APPROVED** [review]

File: web/test/data/private-project.json (hunk 3 above)
Comment (webkhushboo) [reply web/test/data/private-project.json:8]: okay

File: web/test/integration/project.feature (hunk 4 above)
Comment (webkhushboo) [reply web/test/integration/project.feature:14]: ok

**Pull request merged**